src/wallet/api/wallet.cpp

```diff
@@ -327,9 +331,7 @@ bool WalletImpl::close()
     bool result = false;
     LOG_PRINT_L3("closing wallet...");
     try {
-        // do not store wallet with invalid status
-        if (status() == Status_Ok)
-            m_wallet->store();
+        m_wallet->store();
```

Comment:
Status_Error is set when daemon isn't connected -> cache doesn't save on close. But I should add another status code for the daemon instead and just modify this check.